### PR TITLE
[New] Add turndown@4.0.1 w/ npm auto-update

### PR DESCRIPTION
**Pull request by finch185277**
Pull request for issue: #12397

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/domchristie/turndown
 * Official website (optional, not the repository):https://domchristie.github.io/turndown/
 * NPM package url (optional):https://www.npmjs.com/package/turndown
 * License and its reference:MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers:79
   - Count of stars:2662
   - Count of forks:444
 * NPM download stats (optional):
   - Downloads in the last day:209
   - Downloads in the last week:2743
   - Downloads in the last month:4811

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/turndown/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/domchristie/turndown.git"
+  },
+  "npmName": "turndown",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "turndown*"
```
Comment (sufuf3) [inline ajax/libs/turndown/package.json]: As usual, we grab all the files in `dist` & `build` as the basePath, so in this case, we have to use `**/*` as the pattern.
If some files we don't need to fetch when the basePath is `dist` or `build`, we use excluded mode.
So, would you please revise this pattern?
Many thanks.

File: ajax/libs/turndown/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "turndown",
+  "filename": "turndown.min.js",
+  "description": "A library that converts HTML to Markdown",
+  "version": "4.0.1",
+  "author": "Dom Christie",
+  "homepage": "https://domchristie.github.io/turndown/",
+  "keywords": [
+    "converter",
+    "html",
+    "markdown"
```
Comment (sufuf3) [inline ajax/libs/turndown/package.json]: Would you please also add the keywords from topic of [GitHub repository](https://github.com/domchristie/turndown)?
Thanks

**Comment by extend1994** [comment]
In my view, we can't update `to-markdown` to `turndown` because there is a problem: For old versions they were known and used as `to-markdown`, when we update library name we also need to update folder name, which breaks the web pages using the old versions. I can't be sure how if I am correct. Wait for @PeterDaveHello's suggestion.

**Comment by sufuf3** [comment]
I mean that we can use `to-markdown`  as lib name, but update the auto-update config.
However, I not sure.

**Comment by sufuf3** [comment]
@PeterDaveHello Would you please help tell us about this lib, as https://github.com/cdnjs/cdnjs/pull/12430#pullrequestreview-93807556 and https://github.com/cdnjs/cdnjs/pull/12430#issuecomment-364067444 said?
Because the url https://github.com/domchristie/to-markdown always redirect to https://github.com/domchristie/turndown.
Not sure this lib should be added as the new lib, or just update the origin lib.

Many thanks,

**Comment by extend1994** [comment]
ping @PeterDaveHello for this high priority PR.

**Comment by sashberd** [comment]
@PeterDaveHello This PR is waiting for your answer

**Comment by sashberd** [comment]
@MattIPv4 I need your help here. See discussion

**Comment by MattIPv4** [comment]
I think simply treating this as a new library being added is the simplest solution here.

**Pull request merged**